uberfire-security/uberfire-security-api/src/main/java/org/uberfire/security/impl/authz/DefaultAuthorizationPolicy.java

```diff
@@ -30,32 +31,34 @@
 public class DefaultAuthorizationPolicy implements AuthorizationPolicy {
 
     DefaultAuthorizationEntry defaultEntry = new DefaultAuthorizationEntry();
-    private Set<DefaultAuthorizationEntry> entrySet = new HashSet<>();
+    final Map<Group, DefaultAuthorizationEntry> groupEntryMap = new HashMap<>();
+    final Map<Role, DefaultAuthorizationEntry> rolesEntryMap = new HashMap<>();
```

Comment:
```suggestion
    private final Map<Group, DefaultAuthorizationEntry> groupEntryMap = new HashMap<>();
    private final Map<Role, DefaultAuthorizationEntry> rolesEntryMap = new HashMap<>();
```

Reply:
I have added test case covering cache functionality and recalculation of the permissions . Thanks @caponetto for your review and help this. Much appreciated :)